File: app/src/main/java/com/donald/musictheoryapp/Music/ScoreView/ScoreStamp.java
package com.donald.musictheoryapp.Music.ScoreView;

import android.graphics.Canvas;
import android.graphics.Paint;

abstract class ScoreStamp
{
	private final ScoreView m_View;

	ScoreStamp(ScoreView view)
	{
		m_View = view;
	}

	// TODO: MAKE ABSTRACT

	protected ScoreView view() { return m_View; }
	protected float glyphSize() { return m_View.glyphSize(); }
	protected float noteWidth() { return m_View.noteWidth(); }
	protected float staffHeight() { return m_View.staffHeight(); }
	protected float staffStepHeight() { return m_View.staffStepHeight(); }
	protected Paint textPaint() { return m_View.textPaint(); }
	protected Paint glyphPaint() { return m_View.glyphPaint(); }
	protected Paint linePaint() { return m_View.linePaint(); }

	protected abstract void onDraw(Canvas canvas, float posX, float posY);
}

File: app/src/main/java/com/donald/musictheoryapp/customview/scoreview/ScoreStamp.kt
package com.donald.musictheoryapp.customview.scoreview

import android.graphics.Canvas
import android.graphics.Paint

abstract class ScoreStamp(private val view: ScoreView) {

    protected val glyphSize: Float
        get() = view.glyphSize

    protected val noteWidth: Float
        get() = view.glyphSize * REL_SOLID_NOTE_WIDTH

    protected val staffHeight: Float
        get() = view.glyphSize * REL_STAFF_HEIGHT

    protected val staffStepHeight: Float
        get() = view.glyphSize * REL_STAFF_STEP_HEIGHT

    protected val textPaint: Paint
        get() = view.textPaint

    protected val glyphPaint: Paint
        get() = view.glyphPaint

    protected val linePaint: Paint
        get() = view.linePaint

    abstract val relContentWidth: Float

    var relLeftMargin = 0F

    var relRightMargin = 0F

    val leftMargin
        get() = relLeftMargin * glyphSize

    val rightMargin
        get() = relRightMargin * glyphSize

    val relWidth: Float
        get() = relLeftMargin + relContentWidth + relRightMargin

    val width: Float
        get() = /*relContentWidth * glyphSize*/ relWidth * glyphSize

    open fun onDraw(canvas: Canvas, posX: Float, posY: Float) {}

    open fun invalidate() {}
}